src/keys/slip10.rs

```diff
@@ -7,79 +7,168 @@ use alloc::vec::Vec;
 use core::convert::TryFrom;
 
 use serde::{Deserialize, Serialize};
-use zeroize::{Zeroize, ZeroizeOnDrop, Zeroizing};
+use zeroize::{Zeroize, ZeroizeOnDrop};
 
 use crate::macs::hmac::HMAC_SHA512;
-#[cfg(feature = "ed25519")]
-use crate::signatures::ed25519;
-#[cfg(feature = "secp256k1")]
-use crate::signatures::secp256k1_ecdsa;
 
 // https://github.com/satoshilabs/slips/blob/master/slip-0010.md
 // https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki
 // https://en.bitcoin.it/wiki/BIP_0039
 
-#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
-pub enum Curve {
-    #[cfg(feature = "ed25519")]
-    Ed25519,
-    #[cfg(feature = "secp256k1")]
-    Secp256k1,
+pub mod hazmat {
+    pub trait Derivable {
+        fn is_key_valid(key_bytes: &[u8; 33]) -> bool;
+        fn into_key(key_bytes: &[u8; 33]) -> Self;
```

Comment:
I'm nitpicking a bit but Rust usual naming convention for `into_` is that it consumes the parameter, otherwise it's a `to_`

Reply:
Noted.